src/Signal/Window.cc

```diff
@@ -95,26 +117,40 @@ const Core::ParameterBool WindowNode::paramFlushBeforeGap(
 
 WindowNode::WindowNode(const Core::Configuration& c)
         : Component(c), Predecessor(c) {
-    setWindowFunction(WindowFunction::create((WindowFunction::Type)WindowFunction::paramType(c)));
+    setWindowFunction(std::unique_ptr<WindowFunction>(WindowFunction::create((WindowFunction::Type)WindowFunction::paramType(c))));
```

Comment:
```suggestion
    setWindowFunction(std::unique_ptr<WindowFunction>(WindowFunction::create(static_cast<WindowFunction::Type>(WindowFunction::paramType(c)))));
```
C++ style cast instead of C style cast.